File: Java/PrisonerSimulator/src/Player/Strategy/InconsistentStrategy.java
package Player.Strategy;

import Game.Decision;
import Player.IPlayerStrategyFacade;

import java.util.List;
import java.util.Optional;

public class InconsistentStrategy implements IStrategy {
    @Override
    public Decision MakeDecision(IPlayerStrategyFacade player, int rivalId) {
        var myDecisions = player.getMyDecisionsByRivalId(rivalId);
        if (Optional.ofNullable(myDecisions).map(List::isEmpty).orElse(true)) return Decision.Cooperate;

        return myDecisions.get(myDecisions.size() - 1) == Decision.Cooperate ? Decision.Defect : Decision.Cooperate;
    }
}

File: Kotlin/PrisonerSimulator/src/player/strategy/InconsistentStrategy.kt
package player.strategy

import game.Decision
import player.IPlayerStrategyFacade

class InconsistentStrategy : IStrategy {
    override fun makeDecision(player: IPlayerStrategyFacade, rivalId: Int): Decision =
        player.myDecisionsByRivalId[rivalId]?.lastOrNull()?.getOpposite() ?: Decision.Cooperate

}
